### PR TITLE
[LAKESIDE] FLS-92

### PR DESCRIPTION
**Pull request by dyudyunov**
**Description:**
Shippable products rendering fix.
Fix for /about page access by direct link for shippable products.
Product info added to courseware_index.

 Describe in a couple of sentences what this PR adds

**Youtrack:** https://youtrack.raccoongang.com/issue/FLS-92

**Post merge:**
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

File: lms/djangoapps/courseware/views/views.py [hunk 1]
```diff
@@ -880,6 +880,9 @@ def course_about(request, course_id):
         }
         course_tpl = 'courseware/course_about.html'
         product_tpl = 'courseware/product_about.html'
+        # To prevent access to course/about for shippable products by direct link:
+        if course.product and not course.downloadable:
+            raise Http404()
```
Comment (drtwisted) [inline lms/djangoapps/courseware/views/views.py]: Maybe we should better redirect the user to index page?

Comment (dyudyunov) [reply lms/djangoapps/courseware/views/views.py]: @cmltaWt0 what do you think?

File: lms/static/js/index_discovery/models/index_course_card.js [hunk 2]
```diff
@@ -16,7 +16,10 @@
                 start: '',
                 image_url: '',
                 org: '',
-                id: ''
+                id: '',
+                product: false,
+                downloadable: false,
+                external_product_link: ''
```
[the remark this answers is not in the document]
Comment (cmltaWt0) [reply lms/static/js/index_discovery/models/index_course_card.js]: Hmm. Don't want to do it now in JS. Ну его.

File: lms/templates/index_discovery/course_card.underscore [hunk 3]
```diff
@@ -1,9 +1,20 @@
+<%
+    if (product && !downloadable) {
+        var link = external_product_link;
+        var target = '_blank';
+        var buttonText = gettext('BUY NOW');
+    } else {
+        var link = '/courses/' + course + '/about';
```
Comment (cmltaWt0) [inline lms/templates/index_discovery/course_card.underscore]: To refactor - use reverse.
@dyudyunov

File: lms/templates/discovery/course_card.underscore [hunk 4]
```diff
@@ -1,9 +1,20 @@
+<%
+    if (product && !downloadable) {
+        var link = external_product_link;
+        var target = '_blank';
+        var buttonText = gettext('BUY NOW');
+    } else {
+        var link = '/courses/' + course + '/about';
```
Comment (cmltaWt0) [inline lms/templates/discovery/course_card.underscore]: To refactor - use reverse.
@dyudyunov

**Pull request merged**